### PR TITLE
Make Jetpack AI Paragraph block editable

### LINKED ISSUE #28384: [Jetpack AI] Work better with lists and fix styling
Jetpack AI Paragraph does not keep formatting passed in from OpenAI. Even then line breaks scramble upon saving.

The scope of this task is to:
- Transform the plaintext output from OpenAI to add proper paragraphs, lists and (maybe) code blocks
- Insert those as nested blocks or rawhtml inside our block
- Make sure the transformer makes sense. Using the current transformer to paragraph block nests code inside which makes no sense

### Markdown

I tried the following change, and it worked to an extent. The code was transformed, but I didn't know how to write the save function to save this markdown properly.

```
import MarkdownIt from 'markdown-it';
```

```
		.then( res => {
			setLoadingCompletion( false );
			const content = res.prompts[ 0 ].text;
			// This is to animate text input. I think this will give an idea of a "better" AI.
			// At this point this is an established pattern.
			const tokens = content.split( ' ' );
			const md = new MarkdownIt();

			console.log( md.render( content ) );
			for ( let i = 0; i < tokens.length; i++ ) {
				const output = tokens.slice( 0, i ).join( ' ' ).trim();
				setTimeout( () => setAttributes( { content: md.render( output ) } ), 50 * i );
			}
			setTimeout( () => setAttributes( { content: md.render( content.trim() ) } ), 50 * tokens.length );
		} )
```

### PR DESCRIPTION
**Pull request by romarioraffington**
Fixes https://github.com/Automattic/jetpack/issues/28384

Here's a video explaining proposed changes and possible regressions and improvements: 

https://www.loom.com/share/5993f814ab7b4edfb1b2f23ac004391c

### Proposed changes:

* Make the AI Paragraph block editable after AI content is generated

### Testing instructions:

* Add the AI Paragraph block to post
* Edit the content of the AI content that was generated
* Update or publish the post and verify the content is displayed as expected

### Possible Regression / Improvements

1. ~Typing forward slash `/` in the AI Paragraph block should display the block list. Once a block is chosen a new block should be created below the AI Paragraph block~ This is the typical behavior for blocks  ( except for the Paragraph block )
2. ~Creating an AI Paragraph block after another displays cached content.~ This was a cache issue 

### Does this pull request change what data or activity we track or use?
No

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/ai-paragraph/index.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ export const settings = {
 	edit,
 	save: attrs => {
 		const blockProps = useBlockProps.save();
-		return <RawHTML { ...blockProps }>{ attrs.attributes.content }</RawHTML>;
+		return <RichText.Content { ...blockProps } value={ attrs.attributes.content } />;
```
Comment (n3f) [inline projects/plugins/jetpack/extensions/blocks/ai-paragraph/index.js]: I would wrap this in `p` tags
```suggestion
		return <p {...blockProps}><RichText.Content value={ attrs.attributes.content } /></p>;
```

Comment (n3f) [reply projects/plugins/jetpack/extensions/blocks/ai-paragraph/index.js]: This should make the HTML look "better".

**Review by n3f — DISMISSED** [review]
Works for me.  Great work.

**Review by n3f — COMMENTED** [review]
Found an odd bug.  ~~When there is no title~~ the content "flashes".

Kapture 2023-01-26 at 16 39 00

Update:

It occurs with a title too.

**Comment by n3f** [comment]
Also, not your problem, but this doesn't look correct:

image

**Comment by romarioraffington** [comment]
> Found an odd bug. ~When there is no title~ the content "flashes".
> 
> Kapture 2023-01-26 at 16 39 00
> 
> Update:
> 
> It occurs with a title too.

Fixed in [7e9a8c8](https://github.com/Automattic/jetpack/pull/28575/commits/7e9a8c87ff0eefceb196f78e6e69a9f76262bc10)

**Review by n3f — DISMISSED** [review]

File: projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js [hunk 2]
```diff
@@ -291,12 +295,10 @@ export default function Edit( { attributes, setAttributes, clientId } ) {
...
 
 			{ contentIsLoaded && (
-				<ShowLittleByLittle
-					showAnimation={ ! attributes.animationDone }
-					onAnimationDone={ () => {
-						setAttributes( { animationDone: true } );
-					} }
+				<ShowContent
 					html={ attributes.content }
+					setAttributes={ setAttributes }
```
Comment (millerf) [inline projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js]: I think that is a very bad way of doing it, you should rely on a more "React" way and create a "onChange" (and reinstate "onAnimationDone"). This way the component something passed by the parent.

**Review by millerf — COMMENTED** [review]
LGTM. I would prefer not to have setAttributes passed down to "ShowContent", though...

File: projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js (hunk 2 above)
Comment (romarioraffington) [reply projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js]: Fixed in https://github.com/Automattic/jetpack/pull/28575#:~:text=main%20render%20fn-,2b770d6,-romarioraffington%20dismissed%20n3f

**Comment by romarioraffington** [comment]
> LGTM. I would prefer not to have setAttributes passed down to "ShowContent", though...

Fixed in https://github.com/Automattic/jetpack/pull/28575#:~:text=main%20render%20fn-,2b770d6,-romarioraffington%20dismissed%20n3f

**Pull request closed without merging**